### DISCUSSION TITLE
How to use storybook with pinia

### DISCUSSION THREAD

**Discussion by Stygmates** [storybookjs/storybook#17685] [Help] [answered]
Good afternoon,
As Pinia is now the [recommended store](https://pinia.vuejs.org/introduction.html#comparison-with-vuex) for vue3, I can't find examples on how to use it with storybook.
Did someone manage to make it work?

**Comment by romanslonov** [comment, marked as answer]
It's simple. Just init pinia store in `.storybook/preview.js` as you usually do it in vue app (v3).

```js
// .storybook/preview.js

import { app } from '@storybook/vue3';
import { createPinia } from 'pinia';

const pinia = createPinia();

app.use(pinia);

**Reply by iambumblehead** [reply]
how do we attach different pinia states storybook controls?

**Reply by elenakout** [reply]
Is there an example for attaching different pinia states to storybook controls?

**Reply by mukhrr** [reply]
as far as I know [docs ](https://storybook.js.org/recipes/pinia#2-register-pinia) has changed since this answer

**Reply by JoJk0** [reply]
Docs don't include the manipulation of the state inside Pinia within component. They only include workaround to decouple Pinia into parent component and pass the state to the child through props. This is not ideal as for large applications, it involves a lot of prop drilling.

**Reply by Andr1yk0** [reply]
@JoJk0 Yeah, I don't think people use the example from the docs in real applications. 
I've created a decorator and use it in my stories if components need state, works great for me. 
``` js
import { createPinia, setActivePinia } from 'pinia'

export default (story) => {
  setActivePinia(createPinia())
  return story()
}
```
I've created a short post in my blog to describe how it works: [How to use Pinia with Storybook](https://prostocode.com/posts/how-to-use-pinia-with-storybook)

**Comment by Stygmates** [comment]
Storybook documentation has been updated to show how to use pinia with storybook: https://storybook.js.org/tutorials/intro-to-storybook/vue/en/data/
